packages/search-in-workspace/src/browser/search-in-workspace-result-tree-widget.tsx

```diff
@@ -637,6 +646,17 @@ export class SearchInWorkspaceResultTreeWidget extends TreeWidget {
         </React.Fragment>;
     }
 
+    /**
+     * Get the editor widget by the node.
+     * @param {SearchInWorkspaceResultLineNode} node - the node representing a match in the search results.
+     * @returns The editor widget to which the text replace will be done.
+     */
+    protected async doGetWidget(node: SearchInWorkspaceResultLineNode): Promise<EditorWidget> {
+        const fileUri = new URI(node.fileUri);
+        const editorWidget = await this.editorManager.getOrCreateByUri(fileUri);
```

Comment:
if it creates a new widget, at what point of time is gets disposed and garbage collected?

Reply:
I thought it would be handled as it extends `widget`. Could you please tell me how I can do memory leak check for Theia?

Reply:
> if it creates a new widget, at what point of time is gets disposed and garbage collected?
> 
> Could you check that there are no memory leaks with this change please in the case if this code creates a new instance.

I thought it would be handled as it extends widget. Could you please tell me how I can do memory leak check for Theia?

Reply:
I added a bunch of code inside this function and displayed all the editors tracked by the editor manager, it happens that editors are leaking indeed. Good thing is that we only leak one of the same URI, since `getOrCreate` won't spawn duplicates.

I don't know how you should go about disposing them, I also find weird that we have to instantiate `EditorWidget` in order to change a file and save it?

@akosyakov ?

Reply:
> I thought it would be handled as it extends widget. Could you please tell me how I can do memory leak check for Theia?

In dev tools, there is memory tab, you should force garbage collection first, then take a snapshot, and check whether there are no `EditorWidget`, except a constructor (you will see a special character next to an instance). 

> I don't know how you should go about disposing them, I also find weird that we have to instantiate EditorWidget in order to change a file and save it?

You should use `ResourceProvider` to get a `Resource` and then modify and dispose it if you want to modify content on the disk without editor or dirty model.

Reply:
In the updated patch, I manually dispose the editor widget after it has been modified and saved, since editor widget has to be called in order to use the `replaceText` method. I tested with dev tools and could confirm that the editor widget could be disposed properly.

Reply:
please ask someone to verify that:
- it does not cause some glitches, like this editor also gets really opened in the meantime and then unexpectedly closed
- there is no memory leak

If someone checks it and it is fine then it is fine to merge. Don't have time to do it now sorry

Reply:
I looked and did not find a memleak, approving :)